### PR TITLE
ENH: allow using '+' sign in the argument to `to_offset()`

### PR DESCRIPTION
**Pull request by frexvahi**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by frexvahi** [comment]
I made the change here in frequencies.pxd, which presumably means that now frequencies can be specified with a '+' sign. Maybe this is not desired, in which case I could include a modified version of the `opattern` regexp in the to_offset function itself.

**Comment by frexvahi** [comment]
Should the whatsnew entry go in "Other API changes"?

**Comment by frexvahi** [comment]
Also, does this need a separate issue?

File: pandas/tseries/frequencies.py [hunk 1]
```diff
@@ -445,8 +445,8 @@ def to_offset(freq):
                     delta = offset
                 else:
                     delta = delta + offset
-        except Exception:
-            raise ValueError(_INVALID_FREQ_ERROR.format(freq))
+        except Exception as e:
+            raise ValueError(_INVALID_FREQ_ERROR.format(freq)) from e
```
Comment (TomAugspurger) [inline pandas/tseries/frequencies.py]: I don't think this is valid syntax in python 2

File: pandas/tests/tseries/test_frequencies.py [hunk 2]
```diff
@@ -169,6 +169,15 @@ def test_to_offset_leading_zero(self):
         result = frequencies.to_offset(freqstr)
         assert (result.n == -194)
 
+    def test_to_offset_leading_plus(self):
```
Comment (TomAugspurger) [inline pandas/tests/tseries/test_frequencies.py]: Could you add a couple invalid freqstrs, like `'+-1d'`, `'-+1d'`, `'+1'`, `'+d'` and ensure that they all raise.

At least I think they should all raise? `d` is apparently valid, but `-d` isn't.

**Review by TomAugspurger — COMMENTED** [review]
This can go under "Other Enhancements" and you can use 18171 for the issue number. No need for a separate issue.

File: doc/source/whatsnew/v0.22.0.txt [hunk 3]
```diff
@@ -23,7 +23,7 @@ Other Enhancements
 ^^^^^^^^^^^^^^^^^^
 
 - Better support for ``Dataframe.style.to_excel()`` output with the ``xlsxwriter`` engine. (:issue:`16149`)
--
+- ``pd.tseries.frequencies.to_offset()`` now accepts '+' signs e.g. '+1h'. (:issue:`18171`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: use a ``:func:`pandas.tseries.frequencies.to_offset` `` here

say accepts leading '+' signs

**Review by jreback — CHANGES_REQUESTED** [review]
small doc change. ping on green.

File: doc/source/whatsnew/v0.22.0.txt (hunk 3 above)
Comment (frexvahi) [reply doc/source/whatsnew/v0.22.0.txt]: OK, done

File: doc/source/whatsnew/v0.22.0.txt [hunk 4]
```diff
@@ -23,7 +23,7 @@ Other Enhancements
 ^^^^^^^^^^^^^^^^^^
 
 - Better support for ``Dataframe.style.to_excel()`` output with the ``xlsxwriter`` engine. (:issue:`16149`)
--
+- :func:`pd.tseries.frequencies.to_offset()` now accepts leading '+' signs e.g. '+1h'. (:issue:`18171`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: I think this needs to be pandas (and not pd)

Comment (frexvahi) [reply doc/source/whatsnew/v0.22.0.txt]: The other entries in the file all say 'pd' and not 'pandas'. In previous whatsnew.txt files there seems to be a mixture but still a preponderance of 'pd'

Comment (jreback) [reply doc/source/whatsnew/v0.22.0.txt]: hmm, then maybe this works. ok.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @frexvahi

**Pull request merged**